packages/playground/website/src/lib/state/url/resolve-blueprint-from-url.ts

```diff
@@ -215,8 +215,37 @@ function applyQueryOverridesToDeclaration(
 		}
 	}
 
-	// Login
-	if (query.get('login') !== 'no') {
+	const shouldSetLoginToTrue = () => {
+		/**
+		 * Allow the Query API to explicitly set login
+		 * if the login query param is provided.
+		 */
+		if (query.get('login') === 'no') {
+			return false;
+		}
+		if (query.get('login') === 'yes') {
+			return true;
+		}
+
+		/**
+		 * Set login to true by default in the Blueprint
+		 * only if it doesn't already contain a login step or shorthand.
+		 * Otherwise, the login provided by the blueprint would be overridden.
+		 */
+		if (
+			blueprint.steps?.some(
+				(step) =>
+					step && typeof step === 'object' && step?.step === 'login'
+			)
+		) {
+			return false;
+		}
+		if (blueprint.login !== undefined) {
+			return false;
+		}
+		return true;
+	};
+	if (shouldSetLoginToTrue()) {
 		blueprint.login = true;
 	}
```

Comment:
How about this? 

```suggestion
	// Override the `login` property if explicitly requested:
	if (query.get('login') === 'yes') {
		blueprint.login = true;
	} else if(query.get('login') === 'no') {
		blueprint.login = false;
	}
```

Reply:
That's much cleaner, but with it login defaults to `false`. Do we want to change the default login value?

Now, if you go to http://127.0.0.1:5400/website-server/?php=8.3, the user won't be logged in, while before the change, they would be.

Opening Playground without any query arguments would still login the user because we apply the [welcome Blueprint by default](https://github.com/WordPress/blueprints/tree/trunk/blueprints/welcome).

Reply:
Ah, good point! Hmm. I guess we need to know whether `isUserProvidedBlueprint`. Basically, whether the user provided either a Blueprint URL or JSON. With that knowledge, we could express a condition such as:

```ts
if (query.get('login') === 'yes' || !isUserProvidedBlueprint) {
	blueprint.login = true;
} else if // ...
```

What do you think?

Reply:
Does that mean if I:
1. add `"login": false` to my blueprint.json for my plugin SVN; and
2. have my own "login" in "steps" with specific non-admin user.

Then the resulting Live Preview would have the non-admin user login instead of admin?

Reply:
> Then the resulting Live Preview would have the non-admin user login instead of admin?

Unfortunately no. If you add `login: false` to the Blueprint, it would be overridden by the Query API to `true`.
You can test it with [this Playground link.](https://playground.wordpress.net/#{%22landingPage%22:%20%22/wp-admin/profile.php%22,%22extraLibraries%22:%20[%22wp-cli%22],%22login%22:%20false,%22steps%22:%20[{%22step%22:%20%22wp-cli%22,%22command%22:%20%22wp%20user%20create%20user%20user@example.com%20%20--user_pass='password'%20--display_name='Demo%20User'%20--user_nicename='Demo%20User'%20--description='This%20is%20a%20demo%20user.'%22},{%22step%22:%20%22login%22,%22username%22:%20%22user%22,%22password%22:%20%22password%22}]})